scripts/validate_docstrings.py

```diff
@@ -183,20 +182,25 @@ def validate_pep8(self):
             )
         )
 
-        application = flake8.main.application.Application()
-        application.initialize(["--quiet"])
-
+        error_messages = []
         with tempfile.NamedTemporaryFile(mode="w", encoding="utf-8") as file:
             file.write(content)
             file.flush()
-            application.run_checks([file.name])
-
-        # We need this to avoid flake8 printing the names of the files to
-        # the standard output
-        application.formatter.write = lambda line, source: None
-        application.report()
-
-        yield from application.guide.stats.statistics_for("")
+            # GH40784
+            cmd = ["flake8", "--quiet", "--statistics", file.name]
+            response = subprocess.run(cmd, capture_output=True, text=True)
+            stdout = response.stdout
+            # Remove file name from the start
+            stdout = stdout.replace(file.name, "")
+            # Remove the first and last elements since they are always empty str
+            messages = stdout.split("\n")[1:-1]
```

Comment:
does `messages = stdout.strip("\n")` work? If so, let's do that and remove the comment above

Reply:
`messages = stdout.strip("\n")` will fail but I can:

```python
if len(stdout) > 0:
    messages = stdout.strip("\n").split("\n")
```

I am not sure which one is better in terms of readability. What do you think?

Reply:
How about
```python
            messages = stdout.strip("\n")
            if messages:
                error_messages.append(messages)
```
?